runtime/oti/VMHelpers.hpp

```diff
@@ -1364,6 +1364,30 @@ class VM_VMHelpers
 			break;
 		}
 	}
+
+	/**
+	 * @brief Notify JIT first attempt to modified final field of a class
+	 * @param currentThread
+	 * @param fieldClass The declaring class of the final field attempting to modify
+	 */
+	static VMINLINE void
+	reportFinalFieldModified(J9VMThread* currentThread, J9Class* fieldClass) {
+		if (J9_ARE_ANY_BITS_SET(vm->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_OSR_SAFE_POINT)) {
```

Comment:
Maybe flip the order of these checks (and move the jitConfig into a deeper scope).  Already being flagged as illegal is the only condition that is actually likely to occur (JIT off or OSR safe point off are not modes anyone is going to run in practice).